src/gd_avif.c

```diff
@@ -402,7 +402,13 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromAvifCtx (gdIOCtx *ctx)
 	// (While AVIF image pixel depth can be 8, 10, or 12 bits, GD truecolor images are 8-bit.)
 	avifRGBImageSetDefaults(&rgb, decoder->image);
 	rgb.depth = 8;
+#if AVIF_VERSION >= 1000000
```

Comment:
assuming the APIs used exist in older versions, better to write this in C.  it provides better compile-time coverage of code (like typos).

```
if (AVIF_VERSION >= 1000000) {
  ...
} else {
  ...
}
```

Reply:
I don't think this can work, because then the compiler would see an assignment of a return value for some `void` functions.

Reply:
makes sense since `avifRGBImageAllocatePixels` was void before 1.0.0